static/app/components/events/interfaces/spans/anchorLinkManager.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+export type AnchorLinkManagerChildrenProps = {
+  registerScrollFn: (id: string, fn: () => void) => void;
+};
+
+const AnchorLinkManagerContext = React.createContext<AnchorLinkManagerChildrenProps>({
+  registerScrollFn: () => () => undefined,
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export class Provider extends React.Component<Props> {
+  componentDidMount() {
+    this.scrollToHash(location.hash);
+    window.addEventListener('hashchange', () => this.scrollToHash(location.hash), false);
```

Comment:
Window.addEventListener should be paired with a remove.

Reply:
no longer necessary after 2864273